playground/AzureSearchEndToEnd/AzureSearch.AppHost/Program.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddAzureProvisioning();
+
+var azureSearch = builder.AddAzureSearch("search");
+
+builder.AddProject<Projects.AzureSearch_ApiService>("api")
+       .WithReference(azureSearch);
```

Comment:
not a feedback comment, but a question about using the API.
If I don't want to provision, and instead want to have the `api` service connect to an existing `search` resource by specifying an endpoint+key, then should the following work?
```csharp
var builder = DistributedApplication.CreateBuilder(args);
builder.Configuration["ConnectionStrings:search"] = "Endpoint=https://foobar/;ApiKey=YOUR_SEARCH_FOO";
var azureSearch = builder.AddAzureSearch("search");

builder.AddProject<Projects.AzureSearch_ApiService>("api")
       .WithReference(azureSearch);

// This project is only added in playground projects to support development/debugging
// of the dashboard. It is not required in end developer code. Comment out this code
// to test end developer dashboard launch experience. Refer to Directory.Build.props
// for the path to the dashboard binary (defaults to the Aspire.Dashboard bin output
// in the artifacts dir).
builder.AddProject<Projects.Aspire_Dashboard>(KnownResourceNames.AspireDashboard);

builder.Build().Run();
```

This fails with an exception complaining that the connection string named `search` could not be found. But if I add `builder.AddAzureProvisioning()` then it works.

IIUC, `AzureSearchProvisioner.ConfigureResource` populates the connection string when the provisioner is being used.

My expectation here is that:
1. `builder.AddAzureSearch("search");` - this will cause `ConnectionStrings:search` to be tried as a source for the string
2. `builder.AddProject<Projects.AzureSearch_ApiService>.WithReference(azureSearch)` - would mean that the connection string would be passed to the api service

I'm a noob, so I might have some or all of it wrong!

cc @eerhardt @mitchdenny @joperezr

Reply:
@radical you would do the following:

```csharp
var builder = DistributedApplication.CreateBuilder(args);
var azureSearch = builder.AddConnectionString("search");

builder.AddProject<Projects.AzureSearch_ApiService>("api")
       .WithReference(azureSearch);

builder.AddProject<Projects.Aspire_Dashboard>(KnownResourceNames.AspireDashboard);

builder.Build().Run();
```

The in your `secrets.json` you would add the following:

```json
{
  "Parameters": {
    "search": "your azure search connection string"
  }
}
```

When you hit F5 it will read in the connection string from `secrets.json`.